### DISCUSSION TITLE
How to setup Python segment?

### DISCUSSION THREAD

**Discussion by Daksh777** [JanDeDobbeleer/oh-my-posh#856] [Troubleshoot] [answered]
Even after reading the docs, I really can't seem to understand how can I setup the segment for Python.
I think I need to export the config  butI can't seem to find how to export it in the docs either.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@Daksh777 for which platform is this? You can [export the theme](https://ohmyposh.dev/docs/pwsh#override-the-theme-settings) you like (or copy it from here), store that in a file you give your own name and set that as the theme/config to use. You can then easily extend/adjust the [segments](https://ohmyposh.dev/docs/python) to your liking.

**Reply by Daksh777** [reply]
For Windows Terminal.
Okay, so I exported the theme as JSON and added it to the `$profile` and added the sample Python configuration but it seems to be resetting the current theme despite the `Set-PoshPrompt -Theme stelbent.minimal` in the $profile and also, I do not see any changes in the Python segment for some reason?

**Reply by JanDeDobbeleer** [reply]
Can you show me your`$PROFILE`? If I understand correctly, you don't need to add the json to your `$PROFILE`, but create a new file with that content (all [themes](https://github.com/JanDeDobbeleer/oh-my-posh/tree/main/themes) are that anyways), then reference that new file's locations and name from your `$PROFILE`.

Suppose you export it like this: `Export-PoshTheme -FilePath ~/.mytheme.omp.json -Format json`, this creates a new file `.mytheme.omp.json` in your `$HOME` folder. You can then set it (like you do with `stelbent.minimal`) using `Set-PoshPrompt -Theme ~/.mytheme.omp.json`.

**Reply by Daksh777** [reply]
Thank you so much @JanDeDobbeleer, I got it to work.
However, I see extremely long loading times (around 600-700ms). All I have in my `$PROFILE` is `Set-PoshPrompt -Theme ~/terminal.omp.json`

I just added the sample config of the Python to the existing `stelbent.minimal` theme and it increased the loading time. Do you have any idea why would that happen?

**Reply by JanDeDobbeleer** [reply]
No, Powershell is known to be slow to load. If you run that command directly from the CLI, you'll see it does not take 700ms to run. So I can't help there.

**Reply by Daksh777** [reply]
You're correct.
Thank you for the help.
